src/bot.ts

```diff
@@ -523,12 +526,11 @@ you can circumvent this protection against memory leaks.`);
  *
  * @param task Async task to perform
  */
-async function withRetries(task: () => Promise<unknown>) {
-    let success = false;
-    while (!success) {
+async function withRetries<T>(task: () => Promise<T>): Promise<T> {
+    let result: { ok: false } | { ok: true; value: T } = { ok: false };
+    while (!result.ok) {
         try {
-            await task();
-            success = true;
+            result = { ok: true, value: await task() };
         } catch (error) {
             debugErr(error);
             if (error instanceof HttpError) continue;
```

Comment:
> * backoff delay

There's no reason to assume that a network call does much harm if it does not succeed anyway. Also, there is no way to know how long we need to wait. Might as well only wait the time for the network call to time out.

> * max retries

There is no good exit strategy, i.e. we cannot answer the question what to do after the last retry. We must assume that network calls can always fail with a certain probability. If we were to cap the number of retries at `n`, we can no longer reliably perform any operation.

Reply:
yeah, but a network error will infinitely retry immediately, right?

Reply:
If the network is down indefinitely, then yes, grammY will try to reconnect until eternity.

Reply:
Experience shows that calls to `getMe` often fail with a 5xx error. They can be retried successfully and immediately. There is no reason to make the startup of the bot 15 seconds longer than it could be.

It is not acceptable to simply disregard flood wait errors. They must be handled correctly.

Reply:
Besides, the current discussion is unrelated to this PR. Error handling the and API call deduplication are orthogonal. Please leave a comment on #76 if you have further questions or concerns :)